### PR TITLE
Update socket stubs for Python3.12

### PR DESCRIPTION
**Pull request by sobolevn**
Docs: https://github.com/python/cpython/blob/3.12/Doc/library/socket.rst
Source: https://github.com/python/cpython/blob/3.12/Modules/socketmodule.c#L8441

### CODE REVIEW

File: stdlib/_socket.pyi [hunk 1]
```diff
@@ -692,3 +692,28 @@ if sys.platform != "win32" or sys.version_info >= (3, 8):
     def if_nameindex() -> list[tuple[int, str]]: ...
     def if_nametoindex(__name: str) -> int: ...
     def if_indextoname(__index: int) -> str: ...
+
+if sys.version_info >= (3, 12):
```
Comment (AlexWaygood) [inline stdlib/_socket.pyi:696]: Do these need to be re-exported from socket.pyi as well?

Comment (sobolevn) [reply stdlib/_socket.pyi:696]: Yes, because in runtime all is re-exported, good catch!

File: stdlib/_socket.pyi [hunk 2]
```diff
@@ -692,3 +692,28 @@ if sys.platform != "win32" or sys.version_info >= (3, 8):
...
+        HVSOCKET_CONNECT_TIMEOUT: int
+        HVSOCKET_CONNECT_TIMEOUT_MAX: int
+        HVSOCKET_CONNECTED_SUSPEND: int
+        HVSOCKET_ADDRESS_FLAG_PASSTHRU: int
+        HV_GUID_ZERO: int
+        HV_GUID_WILDCARD: int
+        HV_GUID_BROADCAST: int
+        HV_GUID_CHILDREN: int
+        HV_GUID_LOOPBACK: int
+        HV_GUID_PARENT: int
```
Comment (AlexWaygood) [inline stdlib/_socket.pyi:714]: These are all strings:

```pycon
>>> import socket
>>> socket.HV_GUID_ZERO
'00000000-0000-0000-0000-000000000000'
>>> socket.HV_GUID_WILDCARD
'00000000-0000-0000-0000-000000000000'
>>> socket.HV_GUID_BROADCAST
'FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF'
>>> socket.HV_GUID_CHILDREN
'90DB8B89-0D35-4F79-8CE9-49EA0AC8B7CD'
>>> socket.HV_GUID_LOOPBACK
'E0E16197-DD56-4A10-9195-5EE7A155A838'
>>> socket.HV_GUID_PARENT
'A42E7CDA-D03F-480C-9CC2-A4DE20ABB878'
```

```suggestion
        HV_GUID_ZERO: str
        HV_GUID_WILDCARD: str
        HV_GUID_BROADCAST: str
        HV_GUID_CHILDREN: str
        HV_GUID_LOOPBACK: str
        HV_GUID_PARENT: str
```

File: stdlib/socket.pyi [hunk 3]
```diff
@@ -438,6 +438,37 @@ if sys.platform == "win32":
...
+        IP_ADD_SOURCE_MEMBERSHIP as IP_ADD_SOURCE_MEMBERSHIP,
+        IP_BLOCK_SOURCE as IP_BLOCK_SOURCE,
+        IP_DROP_SOURCE_MEMBERSHIP as IP_DROP_SOURCE_MEMBERSHIP,
+        IP_PKTINFO as IP_PKTINFO,
+        IP_UNBLOCK_SOURCE as IP_UNBLOCK_SOURCE,
+    )
+
+    if sys.platform == "win32":
+        from _socket import (
+            AF_HYPERV as AF_HYPERV,
```
Comment (AlexWaygood) [inline stdlib/socket.pyi:452]: In `socket.pyi`, unlike in `_socket.pyi`, this is a member of the `AddressFamily` enum:

```pycon
>>> import socket
>>> socket.AF_HYPERV
<AddressFamily.AF_HYPERV: 34>
```

**Review by AlexWaygood — CHANGES_REQUESTED** [review]

**Comment by hauntsaninja** [comment]
(Let's merge in #10253 first)

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Comment by sobolevn** [comment]
It is much easier with `stubtest`! 👍 
Thanks for adding it.

**Pull request merged**